### ISSUE TITLE
Chauffeur Runner errors with fresh install of Umbraco and Chauffeur

### ISSUE THREAD

**Issue by GarethBlain** [aaronpowell/Chauffeur#65]
I watched your video on "Automating Umbraco with Chauffeur" (https://www.youtube.com/watch?v=CQFu3fmIdms) and wanted to give it a go (we are starting to use Umbraco and deployment is crazy!).

- Using the latest VS 2017 I created a new empty web project using .net 4.7.
- I then added the latest UmbracoCMS from the NuGet manager (7.7.4) and the latest Chauffeur & Chauffeur.Runner (both v0.12.1)
- I rebuilt the project
- Opened a Command Prompt to the bin folder
- Ran `Chauffeur.Runner.exe` and was presented with the following error:

```
Unhandled Exception: System.Reflection.ReflectionTypeLoadException: Could not load all types from "Chauffeur, Version=0.12.0.0, Culture=neutral, PublicKeyToken=null" due to LoaderExceptions, skipping:
. System.IO.FileNotFoundException: Could not load file or assembly 'System.ValueTuple, Version=4.0.2.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system cannot find the file specified.
. System.IO.FileNotFoundException: Could not load file or assembly 'System.ValueTuple, Version=4.0.2.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system cannot find the file specified.

   at Umbraco.Core.TypeFinder.GetTypesWithFormattedException(Assembly a)
   at Umbraco.Core.TypeFinder.GetClassesWithBaseType(Type baseType, IEnumerable`1 assemblies, Boolean onlyConcreteClasses, Func`2 additionalFilter)
   at Umbraco.Core.TypeFinder.FindClassesOfType[T](IEnumerable`1 assemblies, Boolean onlyConcreteClasses)
   at Chauffeur.ShittyIoC..ctor()
   at Chauffeur.Host.UmbracoHost..ctor(TextReader reader, TextWriter writer)
   at Chauffeur.Runner.Program.Main(String[] args)
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   at System.AppDomain.ExecuteAssembly(String assemblyFile, String[] args)
   at System.AppDomain.ExecuteAssembly(String assemblyFile, String[] args)
   at Chauffeur.Runner.Program.Main(String[] args)
```

After digging through the msg and testing I established that Chauffeur Runner requires `System.ValueTuple, Version=4.0.2.0` or above. Upon installing System.ValueTuple v4.4.0 (latest) and re-running the `Chauffeur.Runner.exe` command it fixed it.

I couldn't see anything in your docs about this and it wasn't mentioned in the video's I've watched. Also the error message isn't great! Though maybe if it's a requirement that isn't installed with Umbraco (at least the current version) you could add it as a dependency to your Nuget package and/or put it in your docs. ;)

**Comment by aaronpowell** [comment]
Ah damn, sorry @GarethBlain, I forgot that I added `System.ValueTuple` as a dependency in `0.12` as I move to more C# 7 features, I need to add it as a dependency in the `nuspec` file. I'll get on that right now for you 😃

**Comment by aaronpowell** [comment]
Can you try the `0.12.2-nuspec-fix-116` pre-release version of Chauffeur, my testing indicates it fixes the issue but I'd like to confirm with you first 😉 

- [Chauffeur](https://www.nuget.org/packages/Chauffeur/0.12.2-nuspec-fix-116)
- [Chauffeur.Runner](https://www.nuget.org/packages/Chauffeur.Runner/0.12.2-nuspec-fix-116)

**Comment by GarethBlain** [comment]
Seemed to have fixed it.
- New empty web project
- Added UmbracoCms
- Added Chauffeur.Runner specified (it adds the core obvs)
- Tried running the Runner got an error (from **Shitty**IOC... 🤣 ) about missing Umbraco Core... 😬 
- Compiled Project 😝 
- Tried again and Chauffeur Runner worked fine. 😄 

P.S. Is there a forum/faq or way of asking questions on this? I'm sure it's what we need but could do with a bit of help getting stuff going and didn't really want to ask in a ticket but I can't find anything bar the Wiki. Soz! 😄

**Comment by aaronpowell** [comment]
Cool, branch has been merged, a stable `0.12.2` should be published shortly.

Regarding a forum, nah I don't have anything, use our.umbraco, StackOverflow or post them here, I don't mind. If you use our or SO just tweet the link at my so I see it (`@slace` on twitter).
